source/common/router/upstream_request.cc

```diff
@@ -406,6 +406,8 @@ void UpstreamRequest::onPoolFailure(ConnectionPool::PoolFailureReason reason,
     reset_reason = Http::StreamResetReason::ConnectionFailure;
   }
 
+  stream_info_.upstreamInfo()->setUpstreamTransportFailureReason(transport_failure_reason);
+
   // Mimic an upstream reset.
   onUpstreamHostSelected(host);
   onResetStream(reset_reason, transport_failure_reason);
```

Comment:
> I dont understand. The idea here is to propagate the transport failure reason in the context on onPoolFailure. Since the reason is available in the callback onPoolFailure, we need to capture that here.

I mean move the `transport_failure_reason` to the `onResetStream` method here which also can access the `transport_failure_reason`.

Reply:
I see what you are saying. The reason for reset stream reset could be different from connection failure. This PR is for propagating connection failure. I could add the same logic of capturing the failure in `onResetStream` but not sure if that can replace this one.

Reply:
I see, may be you are right.

Reply:
@wbpcode Thanks for the suggestion. Will add the same logic in both places.

Reply:
@wbpcode I have moved the code to `onResetStream` .  One thing I am not sure is - is `onResetStream` only called in  context of upstream error. If not, then we might be introducing a bug where we will see error string in access log which is not upstream related.

Reply:
> I see, may be you are right.

@conqerAtapple Sorry for this 😭. But in fact I think that your previous implementation is right. Could you just revert it? Sorry again.

Reply:
No problem. I will do that. 
What do you think about https://github.com/envoyproxy/envoy/pull/21029#issuecomment-1113528230 ? 

Since the purpose of this PR is to propagate the connection error from upstream (if any) and not adding any new failure message, I think we should not add/change any new failure messages. WDYT?

Reply:
I still think new failure message is ok. But let @alyssawilk make the final decision. Thanks.